TestFoundation/TestFileHandle.swift

```diff
@@ -101,9 +104,28 @@ class TestFileHandle : XCTestCase {
     func createFileHandleForReadErrors() -> FileHandle {
         // Create a file handle where calling read returns -1.
         // Accomplish this by creating one for a directory.
+#if os(Windows)
+        let hDirectory: HANDLE = ".".withCString(encodedAs: UTF16.self) {
+            // NOTE(compnerd) we need the FILE_FLAG_BACKUP_SEMANTICS so that we
+            // can create the handle to the directory.
+            CreateFileW($0, GENERIC_READ,
+                        DWORD(FILE_SHARE_DELETE | FILE_SHARE_READ | FILE_SHARE_WRITE),
+                        nil, DWORD(OPEN_EXISTING),
+                        DWORD(FILE_ATTRIBUTE_NORMAL | FILE_FLAG_BACKUP_SEMANTICS), nil)
+        }
+        if hDirectory == INVALID_HANDLE_VALUE {
+          fatalError("unable to create handle to current directory")
+        }
+        let fd = _open_osfhandle(intptr_t(bitPattern: hDirectory), 0)
+        if fd == -1 {
+          fatalError("unable to associate file descriptor with handle")
```

Comment:
Wouldn't it be better to make this function return a `FileHandle?` so this can `return nil` and  be `XCTFail`ed by the caller, rather than stopping all of the tests?

Reply:
I followed what the old path was doing (fatal error'ing).  I do like that, and since you asked, I'll go ahead and do that.

Reply:
Actually, lets do that in a follow up, I have a huge patch stack locally for the `DEPLOYMENT_TARGET_*` to `TARGET_OS_*` and that is making things more complicated.

Reply:
I took a look at this and this is really ugly without some of the internal helpers (i.e. `_NSErrorWithErrno` and `_NSErrorWithWindowsError`).  Do you want to convert this to a testable import test?  CC: @millenomi

Reply:
Note that `expectTrue()` below doesn't invoke `fatalError()`.

Reply:
I'm not sure if we can do anything else here.  Its an internal error because this implies that the handle that was provided was invalid.  It would indicate an error in the test itself.

Reply:
I'm okay with that.

Reply:
Meaning: I'm okay error'ing out if the text is executing such that '.' is not an accessible directory for reading.